### PR TITLE
Support for copying the cabinet file after the vm creation only in case of windows.

### PR DESCRIPTION
**Pull request by cohesity-mradul**

### CODE REVIEW

File: vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go [hunk 1]
```diff
@@ -109,6 +109,59 @@ func VirtualMachineCustomizeSchema() map[string]*schema.Schema {
 			}},
 		},
 
+		// This is the spec of resource_vsphere_file.To customize a windows machine we
+		// copy the cabinet file inside the VM. This file should be present inside the
+		// vm folder. So when the vm is created we check if there are any customization
+		// for the VM. We do not use the windows_options as it syspreps the machine and
+		// changes the identity of the machine. TODO(Mradul): Take only the source path
+		// and compute the other options from the VM resource definition.
+		"cohesity_windows_customization_otions": {
```
Comment (prakash-cohesity) [inline vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go]: cohesity_windows_customization_options ?

File: vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go [hunk 2]
```diff
@@ -109,6 +109,59 @@ func VirtualMachineCustomizeSchema() map[string]*schema.Schema {
...
+			Type:          schema.TypeList,
+			Optional:      true,
+			MaxItems:      1,
+			ConflictsWith: []string{cKeyPrefix + "." + "windows_options", cKeyPrefix + "." + "windows_sysprep_text", cKeyPrefix + "." + "linux_options"},
+			Description:   "Customization of windows vm in cohesity",
+			Elem: &schema.Resource{Schema: map[string]*schema.Schema{
```
Comment (prakash-cohesity) [inline vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go:128]: Can add a comment here that, this schema is same as VSphereFile resource schema.

**Review by prakash-cohesity — APPROVED** [review]
minor name changes and comments should be added.

File: vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go (hunk 1 above)
Comment (cohesity-mradul) [reply vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go]: Done

File: vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go (hunk 2 above)
Comment (cohesity-mradul) [reply vsphere/internal/vmworkflow/virtual_machine_customize_subresource.go:128]: Done

**Pull request merged**